docs/advanced-topics/dockstore-cli/dockstore-cli-faq.rst

```diff
@@ -0,0 +1,95 @@
+  :local:
+
+How does launching with Dockstore CLI compare with cwltool?
+-----------------------------------------------------------
+
+Under the hood, the Dockstore CLI evokes cwltool to launch CWL workflows. However, it adds additional features. Dockstore CLI has utilities to generate JSON parameter files from
+entries on Dockstore (``dockstore tool convert``).
+
+When launching tools, the Dockstore CLI makes it easy to specify entries
+from the Dockstore website. We can also provision input and output files using HTTP,
+FTP, S3, and GCS. As of Release 1.12, the Dockstore CLI has support for running on `a WES server <https://github.com/ga4gh/workflow-execution-service-schemas>`__. We also have preliminary support for `Synapse <https://www.synapse.org/>`__ and the `ICGC Storage
+client <https://docs.icgc.org/download/guide/#score-client-usage>`__. Please see `file provisioning plugins <https://github.com/dockstore/dockstore-cli/tree/master/dockstore-file-plugin-parent>`__
+for more information on these two file transfer sources.
+
+.. _how-do-i-use-the-dockstore-cli-on-a-mac:
+
+How do I use the Dockstore CLI on a Mac?
+----------------------------------------
+
+See `Docker for Mac <https://docs.docker.com/engine/installation/mac/>`__ for installation information.
+
+.. note:: Docker behaves a bit differently on a
+    `Mac <https://docs.docker.com/docker-for-mac/osxfs/#/namespaces>`__ than
+    on a typical Ubuntu machine. By default the only shared volumes are
+    /Users, /Volumes, /tmp, and /private. Note that /var is not a shared
+    directory (and can't be set as one). ``cwltool`` uses your TMPDIR (the
+    env variable) to setup volumes with docker, which on a Mac can default
+    to a subdirectory of /var. In order to get ``cwltool`` working on your
+    Mac, you need to set your TMPDIR to be under one of the shared volumes
+    in Docker for Mac. You can do this by doing something similar to the
+    following:
+    ::
+
+        export TMPDIR=/tmp/docker_tmp
+
+By default, Docker for Mac allocates fewer resources (CPU, Memory, Swap)
+to containers compared to what is available on your host machine. You
+can change what it allocates using the Docker for Mac GUI under
+``Preferences > Advanced`` as described
+`here <https://docs.docker.com/docker-for-mac/#advanced>`__.
+
+* The default allocation can cause workflows or tools to fail without informing the user with a memory or resource related error message. If you find that your workflow or tool is behaving differently on a Mac compared to a similarly resourced Ubuntu environment, you can try increasing the resources allocated to Docker on the Mac to resolve the discrepancy.
+
+How can I use the Dockstore CLI with Python 2?
+----------------------------------------------
+
+Python 2 support ended in 2020. You can get the Python 2 requirements.txt with ``curl -o requirements.txt "https://dockstore.org/api/metadata/runner_dependencies?python_version=2"``
```

Comment:
If you execute that curl command, it fails. Looking at the code, it looks like we only support `python_version=2` if you add the dockstore client_version, and it has to be 1.7.0 or less: `curl "https://dockstore.org/api/metadata/runner_dependencies?client_version=1.7.0&python_version=2&runner=cwltool&output=text"`

I suggest either getting rid of this section entirely, or changing the answer to be we don't support Python 2.

P.S. I know you didn't change this, you just moved it, but it shows up as green/changed in GitHub, so...

Reply:
Thanks for catching this. Removed it.